doc: fix Writable.write callback description (#31812)
nodejs/node: doc/api/stream.md

supplied `callback` once the data has been fully handled. If an error
occurs, the `callback` *may or may not* be called with the error as its
first argument. To reliably detect write errors, add a listener for the

> `'error'` event. The `callback` method will always be called asynchronously

Reviewer: ```suggestion
`'error'` event. The `callback` is called asynchronously and before `'error'` is emitted.
```

supplied `callback` once the data has been fully handled. If an error
occurs, the `callback` *may or may not* be called with the error as its
first argument. To reliably detect write errors, add a listener for the
`'error'` event. The `callback` is called asynchronously and before `'error'` is
emitted.

> and before `'error'` is emitted.

Reviewer: ```suggestion
```
